apis/outlook/mail/code/pkg/printers/mail.go

```diff
@@ -74,3 +82,47 @@ func PrintMessage(msg models.Messageable, detailed bool) error {
 func recipientableToString(r models.Recipientable) string {
 	return fmt.Sprintf("%s (%s)", util.Deref(r.GetEmailAddress().GetName()), util.Deref(r.GetEmailAddress().GetAddress()))
 }
+
+func fixMailFolder(folder *models.MailFolderable) error {
+	if folder == nil {
+		return nil
+	}
+
+	newFolderID, err := id.SetOutlookID(util.Deref((*folder).GetId()))
+	if err != nil {
+		return fmt.Errorf("failed to set folder ID: %w", err)
+	}
+
+	(*folder).SetId(util.Ptr(newFolderID))
```

Comment:
Hmm I think `folder.SetId` should just work?

Reply:
No clue why, but it wasn't working, which is why I had to do it this way.